### PR TITLE
CV2-3777: expose TiplineRequestType

### PR DESCRIPTION
**Pull request by melsawy**
## Description

- Expose tipline request type in GraphQL
- Added a new fields `report_sent_at` , `report_correction_sent_at` & `smooch_request_type`

References: CV2-3777

## How has this been tested?

Implemented automated tests.

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: app/graph/types/query_type.rb [hunk 1]
```diff
@@ -202,6 +202,18 @@ def dynamic_annotation_field(query:, only_cache: nil)
     end
   end
 
+  field :tipline_request,
+        TiplineRequestType,
+        description:
+          "Information about tipline request, given its id",
+        null: true do
+    argument :id, GraphQL::Types::String, required: true
+  end
```
Comment (caiosba) [inline app/graph/types/query_type.rb:211]: Check permissions.

Comment (melsawy) [reply app/graph/types/query_type.rb:211]: Used same permission for `dynamic_annotation_field` calling.

File: app/models/bot/smooch.rb [hunk 2]
```diff
@@ -947,15 +947,23 @@ def self.send_correction_to_user(data, pm, subscribed_at, last_published_at, act
...
+        field_name = 'report_correction_sent_at'
         self.send_report_to_user(uid, data, pm, lang, 'fact_check_report_updated', self.get_string(:report_updated, lang))
       end
     # First report
     else
+      field_name = 'report_sent_at'
       self.send_report_to_user(uid, data, pm, lang, 'fact_check_report')
     end
+    unless field_name.blank?
+      a = pm.get_annotations('smooch').last.load
```
[the remark this answers is not in the document]
Comment (melsawy) [reply app/models/bot/smooch.rb]: Done

File: app/graph/types/query_type.rb (hunk 1 above)
Comment (caiosba) [reply app/graph/types/query_type.rb:211]: I think that permission only applies to global API keys, so it won't work from the web client for regular users, right?

Comment (caiosba) [reply app/graph/types/query_type.rb:211]: Why we need this API endpoint anyway? We always we want to get the requests under the scope of an item, no?

**Review by caiosba — COMMENTED** [review]
Just added two comments, Sawy. Also, please check if the frontend still works with these changes. If not, let's wait to merge this branch until the backend changes are in place too.

File: app/graph/types/query_type.rb (hunk 1 above)
Comment (caiosba) [reply app/graph/types/query_type.rb:211]: Just talked to Sawy - we can delete this block.

Comment (melsawy) [reply app/graph/types/query_type.rb]: Deleted

**Review by caiosba — APPROVED** [review]
Sawy, please just add a test for the new field:
`Unconvered lines in app/models/concerns/smooch_fields.rb: 81, 82`
Good to merge after that.

**Pull request merged**